File: Ej2.java
package ej2;

import javax.swing.JOptionPane;

class Ej2 {

	public static void main(String[] args) {
		double a, prom, b;
		String res;
		prom = 0;
		
		do {
			a = Double.parseDouble(JOptionPane.showInputDialog("ingrese numero"));
			b = Double.parseDouble(JOptionPane.showInputDialog("ingrese numero"));
			prom = (a + b) / 2;
			JOptionPane.showMessageDialog(null, "proedio " + prom);
			res = JOptionPane.showInputDialog("desea calcular otro?");
		} while (res.equalsIgnoreCase("si"));

	}

}

File: Ej2.js
let a;
let b;
let prom;
let answer;

do {
    a = parseFloat(prompt("ingrese numero"));
    b = parseFloat(prompt("ingrese numero"));
    prom = (a + b) / 2;
    alert("promedio " + prom);
    answer = prompt("desea calcular otro?");
} while (answer.toLowerCase() === "si");
